core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -210,6 +210,16 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     }
   }
 
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long) {
```

Comment:
Should this be called `maybeTruncateCheckpoint`? And similarly for the `LogCleaner` method?